### ISSUE TITLE
missing-timeout-minutes rule fires on steps when job-level timeout-minutes is set

### ISSUE THREAD

**Issue by HikaruEgashira** [ultra-supara/sisakulint#428]
## Summary

`missing-timeout-minutes` rule reports findings on individual steps even when the parent job already has `timeout-minutes` configured. Job-level `timeout-minutes` covers all steps within that job, so step-level findings are false positives.

## Reproduction

```yaml
jobs:
  build:
    runs-on: ubuntu-latest
    timeout-minutes: 15  # ← job-level timeout covers all steps
    steps:
      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
        # sisakulint flags missing-timeout-minutes here — FP
```

`sisakulint` reports `[missing-timeout-minutes]` on the checkout step despite the job having `timeout-minutes: 15`.

## Expected behavior

When a job has `timeout-minutes` set, do not flag `missing-timeout-minutes` on its child steps.

## Impact

In eval testing of the sisakulint skill, a fully hardened workflow (3 jobs, all with job-level timeouts) produced 13 false positive findings, causing the skill to unnecessarily modify the workflow.

**Comment by HikaruEgashira** [comment]
Closing — this is not a bug. Step-level timeout-minutes is a valid defense-in-depth measure per GitHub Actions docs. A single step can consume the entire job timeout, so step-level timeouts are a legitimate hardening recommendation even when job-level is set.
